### DISCUSSION TITLE
在hook不定长参数函数（比如：printf）时，代理函数如何调用原始函数？

### DISCUSSION THREAD

**Discussion by Mr-JingShi** [bytedance/bhook#62] [Q&A] [answered]
不定长参数... 好像没发直接传递！

**Comment by caikelun** [comment, marked as answer]
两种情况：
1. 可变部分的参数最大个数可知：按照最大参数个数来定义proxy和传参，可参考 bytehook_sample 中对 `open` 的 hook。
2. 可变部分的参数最大个数不可知：通过对应的 `va_list` 参数版本的函数调用“原函数”，比如 `print` 的 proxy 中调用 `vprintf`。

**Reply by wustwg** [reply]
像printf这样的函数，如果没有valist的函数，那只能按照最大参数个数来定义，但是这个最大参数个数 不确定啊 ，100个也有可能啊？
